core/primitives/chain_epoch.hpp

```diff
@@ -15,7 +14,7 @@ namespace fc::primitives {
    * @brief epoch index type represents a round of a blockchain protocol, which
    * acts as a proxy for time within the VM
    */
-  using ChainEpoch = libp2p::multi::UVarint;
+  using ChainEpoch = UBigInt;
```

Comment:
hmmm, but still - why we cant use ChainEpoch as argument type in functions which now takes uint64_t

Reply:
uint64_t can be cast to UBigInt, not vice versa.